controller/userinfo.go

```diff
@@ -0,0 +1,71 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/fabric8-services/fabric8-auth/account"
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/application"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/token"
+	"github.com/goadesign/goa"
+)
+
+// UserinfoController implements the userinfo resource.
+type UserinfoController struct {
+	*goa.Controller
+	db           application.DB
+	tokenManager token.Manager
+}
+
+// NewUserinfoController creates a userinfo controller.
+func NewUserinfoController(service *goa.Service, db application.DB, tokenManager token.Manager) *UserinfoController {
+	return &UserinfoController{
+		Controller:   service.NewController("UserinfoController"),
+		db:           db,
+		tokenManager: tokenManager,
+	}
+}
+
+// Show runs the show action.
+func (c *UserinfoController) Show(ctx *app.ShowUserinfoContext) error {
+	id, err := c.tokenManager.Locate(ctx)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "Bad Token")
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("Bad Token"))
```

Comment:
In GET `/user` this is bad request, but I think if token is invalid Unauthorized would me more appropriate. Let me know what do you think. Also I have changed all generic errors (pkg/errors) to autherrors(fabric8-auth/errors)

Reply:
Yes, I think 401 is more appropriate here. Just use lower case and add the original error message:
jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("bad token: " + err.String()))